corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -128,35 +128,38 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
 
         var queryResponse = menuResponse.queryResponse;
         if (sidebarEnabled && menuResponse.type === "entities" && queryResponse)  {
-            let queryCollection = new Collection(queryResponse.displays);
+            var queryCollection = new Collection(queryResponse.displays);
             // hasOwnProperty check only required temporarily for transient moment during deploy when response will not contain groupHeaders
-            if (queryResponse.hasOwnProperty("groupHeaders") && Object.keys(queryResponse.groupHeaders).length > 0) {
-                queryCollection = new Collection(menusUtils.groupDisplays(new Collection(queryResponse.displays), queryResponse.groupHeaders));
+            let myGroupHeaders = {};
+            if (queryResponse.hasOwnProperty("groupHeaders")) {
```

Comment:
I ended up deploying Formplayer to prod already so we will no longer need these checks.